src/core/execution/TradeShipExecution.ts

```
@@ -103,46 +102,81 @@ export class TradeShipExecution implements Execution {
       case PathFindResultType.Pending:
         // Fire unit event to rerender.
-        this.tradeShip.touch();
+        this.tradeShip!.touch();
         break;
-      case PathFindResultType.NextTile:
-        this._dstPort.cachePut(this.tradeShip.tile(), result.tile);
-        // Update safeFromPirates status
-        if (this.mg.isWater(result.tile) && this.mg.isShoreline(result.tile)) {
-          this.tradeShip.setSafeFromPirates();
+      case PathFindResultType.Completed:
+        const fullPath = this.aStar.reconstructPath();
```

Biome (1.9.4)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

The declaration is defined in this switch clause:

Unsafe fix: Wrap the declaration in a block.


(lint/correctness/noSwitchDeclarations)